075-connector-offsets-support.md

```diff
@@ -0,0 +1,270 @@
+This difference is reflected in the approaches outlined in this proposal.
+
+Source connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        // Connector-defined source partition
+      },
+      "offset": {
+        // Connector-defined source offset
+      }
+    }
+  ]
+}
+```
+
+Sink connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "kafka_topic": // Kafka topic
+        "kafka_partition": // Kafka partition
+      },
+      "offset": {
+        "kafka_offset": // Kafka offset
+      }
+    }
+  ]
+}
+```
+
+### New annotation
+
+All three actions will make use of a new annotation called `strimzi.io/connectorOffsets`.
+The possible values will be `list`, `alter` and `reset`.
+
+### Listing offsets
+
+To list the current offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `list`.
+After the annotation is added, on the next reconciliation the operator will fetch the current offsets and create a Kubernetes ConfigMap containing the JSON response.
+The operator will add an owner reference to the Kubernetes ConfigMap pointing to the KafkaConnector resource that was annotated.
+If the ConfigMap already exists the operator will replace the data with the updated data and not add an owner reference.
+The operator will then remove the annotation from the KafkaConnector CR.
+
+If the user wants to see an updated set of offsets they will need to re-annotate the resource.
+
+The name of the ConfigMap the operator creates/updates will be set by the user in the KafkaConnector CR.
+The KafkaConnector CRD will be updated to contain a new property called `listOffsets` that must be set for the `strimzi.io/connectorOffsets=list` annotation to take effect.
+The structure of the `listOffsets` property will be as below, to allow it to be extended in future if required:
+
+```yaml
+listOffsets:
+  toConfigMap:
+    name: my-connector-offsets
+```
+
+The ConfigMap will contain a single entry called `offsets.json` that contains the response received from the GET /connectors/{connector}/offsets endpoint.
+
+For example for a source connector the ConfigMap might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-source-connector-offsets
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            // Connector-defined source partition
+          },
+          "offset": {
+             // Connector-defined source offset
+          }
+        }
+      ]
+    }
+```
+
+For a sink connector it might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-sink-connector-offsets
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            "kafka_topic": "my-topic"
+              "kafka_partition": 2
+          },
+          "offset": {
+            "kafka_offset": 4
+          }
+        }
+      ]
+    }
+```
```

Comment:
both the above ConfigMap examples are missing the "owner reference to the Kubernetes ConfigMap pointing to the KafkaConnector resource that was annotated" you mentioned before. Maybe you can add it?